### PR TITLE
chore: monorepo & add typescript eslint config

### PR DESCRIPTION
**Pull request by kevinmarrec**
Make the repository a lerna monorepo to provide both `eslint-config` & `eslint-config-typescript`.

https://typescript.nuxtjs.org/guide/lint.html

## TODO 
- [x] Move to monorepo
- [x] Edit homepage links in `package.json` of packages
- [x] Update test
- [x] Update README.md with `TypeScript` section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,36 +26,54 @@ $ npm i -D eslint
...
-3. Create a `.eslintrc` file
+3. Create a `.eslintrc.js` file
 
 4. Extend our config (you can use just the scope name as ESLint will assume the `eslint-config` prefix):
 
-```json
-{
-  "extends": [
-    "@nuxtjs"
+```js
```
Comment (pi0) [inline README.md]: Any objection to prefer `.js`?

File: packages/eslint-config-typescript/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (pi0) [inline packages/eslint-config-typescript/package.json]: Would you please sort it using `npx sort-package-json`?

File: tests/basic.test.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const config = require('../')
+const config = require('../packages/eslint-config')
```
Comment (pi0) [inline tests/basic.test.js]: Can we move tests to the package itself?

File: README.md (hunk 1 above)
Comment (kevinmarrec) [reply README.md]: I don't know, I've been able to see both.
i.e. Http module use it, but PWA module use `.eslintrc`

BUT I just checked and forgot that `.eslintrc` has Code editor support so it fits better
I should rollback these changes

File: packages/eslint-config-typescript/package.json (hunk 2 above)
Comment (kevinmarrec) [reply packages/eslint-config-typescript/package.json]: sure

File: tests/basic.test.js (hunk 3 above)
Comment (kevinmarrec) [reply tests/basic.test.js]: Yeah surely

File: README.md (hunk 1 above)
Comment (pi0) [reply README.md]: (personal opinion) I prefer JSON format as it is simpler and has less boilerplate but js allows more advanced features like custom configuration based on env, etc. IDE support is little more for js (vim, ...) -- just asking to know if there is special typescript requirement. I'm OK with both :)

/cc @manniL @clarkdo

Comment (kevinmarrec) [reply README.md]: Okay, same I'm ok with both, so I reverted so that this PR is lighter, we'll maybe update README afterwards

**Pull request merged**